src/python/WMComponent/ErrorHandler/ErrorHandlerPoller.py

```diff
@@ -233,8 +238,13 @@ def readFWJRForErrors(self, jobList):
                     logging.debug(msg)
                     exhaustJobs.append(job)
                     continue
-
-                if len([x for x in report.getExitCodes() if x in self.exitCodes]):
+                
+                if self.reqAuxDB is None:
+                    exitCodesNoRetry = self.config.ErrorHandler.failureExitCodes
+                else:
+                    exitCodesNoRetry = self.reqAuxDB.getWMAgentConfig(self.config.Agent.hostName).get("NoRetryExitCodes", [])
+                
+                if len([x for x in report.getExitCodes() if x in exitCodesNoRetry]):
```

Comment:
getWMAgentConfig is using MemoryCache class which mean it won't though the exception if call fails it will just return the previous result.